explainability-service/src/main/java/org/kie/trustyai/service/data/storage/hibernate/HibernateStorage.java

```diff
@@ -0,0 +1,587 @@
+
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.transaction.Transactional;
+
+@LookupIfProperty(name = "service.storage.format", stringValue = "DATABASE")
+@ApplicationScoped
+public class HibernateStorage extends Storage<Dataframe, StorageMetadata> {
+    @PersistenceContext
+    EntityManager em;
+
+    private static final Logger LOG = Logger.getLogger(HibernateStorage.class);
+
+    private final int batchSize;
+    private Optional<MigrationConfig> migrationConfig = Optional.empty();
+    private boolean dataDirty = false;
+    private final String NO_DATA_ERROR_MSG = "No inference data for that model found in database.";
+
+    private Random rng = new Random(0);
+    private long updateHash;
+
+    public HibernateStorage(ServiceConfig serviceConfig, StorageConfig storageConfig) {
+        LOG.info("Starting Hibernate storage consumer.");
+
+        if (serviceConfig.batchSize().isPresent()) {
+            this.batchSize = serviceConfig.batchSize().getAsInt();
+        } else {
+            final String message = "Missing data batch size";
+            LOG.error(message);
+            throw new IllegalArgumentException(message);
+        }
+
+        if (storageConfig.dataFilename().isPresent() && storageConfig.dataFolder().isPresent()) {
+            migrationConfig = Optional.of(new MigrationConfig(storageConfig.dataFolder().get(), storageConfig.dataFilename().get()));
+        }
+    }
+
+    @Override
+    public DataFormat getDataFormat() {
+        return DataFormat.HIBERNATE;
+    }
+
+    // MIGRATIONS ======================================================================================================
+    @PostConstruct
+    @Transactional
+    protected void migrate() {
+        if (migrationConfig.isPresent()) {
+            MigrationConfig mc = migrationConfig.get();
+            String fromFolder = mc.getFromFolder();
+            String fromFile = mc.getFromFilename();
+
+            if (!Paths.get(fromFolder).toFile().isDirectory()) {
+                LOG.warn("A storage folder \"" + fromFolder + "\" was provided in the service configuration, but no such directory exists. Migration will be skipped.");
+                return;
+            }
+
+            CustomStorageConfig customStorageConfig = new CustomStorageConfig(fromFile, fromFolder);
+            CustomServiceConfig customServiceConfig = new CustomServiceConfig(OptionalInt.of(batchSize), null, null);
+            PVCStorage pvcStorage = new PVCStorage(customServiceConfig, customStorageConfig);
+            CSVDataSource oldDataSource = new CSVDataSource();
+            oldDataSource.setParser(new CSVParser());
+            oldDataSource.setStorageOverride(pvcStorage);
+            List<String> modelIds = pvcStorage.listAllModelIds();
+
+            if (!modelIds.isEmpty()) {
```

Comment:
IIUC two (more) consecutive migrations will add the PVC data to the already existing data on the DB, duplicating it.
Should we had a check and a rule that if the target DB already has data (and the TrustyAI schemas) the migration should not proceed?